diff --git a/openssl-sys/build/main.rs b/openssl-sys/build/main.rs
index bbee7c5c..57033b2d 100644
--- a/openssl-sys/build/main.rs
+++ b/openssl-sys/build/main.rs
@@ -293,6 +293,7 @@ See rust-openssl documentation for more information:
             (3, 8, 0) => ('3', '8', '0'),
             (3, 8, 1) => ('3', '8', '1'),
             (3, 8, _) => ('3', '8', 'x'),
+            (3, 9, _) => ('3', '8', 'x'),
             _ => version_error(),
         };
 
