diff --git a/openssl-sys/build/main.rs b/openssl-sys/build/main.rs
index cdea3eb4..b16ebecf 100644
--- a/openssl-sys/build/main.rs
+++ b/openssl-sys/build/main.rs
@@ -282,7 +282,7 @@ See rust-openssl documentation for more information:
             (3, 5, _) => ('3', '5', 'x'),
             (3, 6, 0) => ('3', '6', '0'),
             (3, 6, _) => ('3', '6', 'x'),
-            (3, 7, 0) => ('3', '7', '0'),
+            (3, 7, _) => ('3', '7', 'x'),
             _ => version_error(),
         };
 
