diff --git a/openssl-sys/build/main.rs b/openssl-sys/build/main.rs
index ba149c17..5274dc44 100644
--- a/openssl-sys/build/main.rs
+++ b/openssl-sys/build/main.rs
@@ -285,7 +285,7 @@ See rust-openssl documentation for more information:
             (3, 7, 0) => ('3', '7', '0'),
             (3, 7, 1) => ('3', '7', '1'),
             (3, 7, _) => ('3', '7', 'x'),
-            _ => version_error(),
+            _ => ('3', '7', 'x'),
         };
 
         println!("cargo:libressl=true");
