--- crypto/arch/aarch64/crypto_arch.h.orig	Sun Jun 29 03:49:56 2025
+++ crypto/arch/aarch64/crypto_arch.h	Sun Jun 29 04:00:13 2025
@@ -34,11 +34,13 @@ extern uint64_t crypto_cpu_caps_aarch64;
 #define CRYPTO_CPU_CAPS_AARCH64_SHA3	(1ULL << 5)
 
 #ifndef OPENSSL_NO_ASM
+#if 0
 
 #define HAVE_SHA1_BLOCK_DATA_ORDER
 #define HAVE_SHA256_BLOCK_DATA_ORDER
 #define HAVE_SHA512_BLOCK_DATA_ORDER
 
+#endif
 #endif
 
 #endif
