--- crypto/aes/aes_amd64.c.orig	Mon Jun 16 11:48:01 2025
+++ crypto/aes/aes_amd64.c	Mon Jun 16 11:51:51 2025
@@ -45,6 +45,7 @@ void aesni_decrypt(const unsigned char *in, unsigned c
 void aesni_cbc_encrypt(const unsigned char *in, unsigned char *out,
     size_t len, const AES_KEY *key, unsigned char *ivec, const int enc);
 
+#ifdef HAVE_AES_SET_ENCRYPT_KEY_INTERNAL
 int
 aes_set_encrypt_key_internal(const unsigned char *userKey, const int bits,
     AES_KEY *key)
@@ -54,7 +55,9 @@ aes_set_encrypt_key_internal(const unsigned char *user
 
 	return aes_set_encrypt_key_generic(userKey, bits, key);
 }
+#endif
 
+#ifdef HAVE_AES_SET_DECRYPT_KEY_INTERNAL
 int
 aes_set_decrypt_key_internal(const unsigned char *userKey, const int bits,
     AES_KEY *key)
@@ -64,7 +67,9 @@ aes_set_decrypt_key_internal(const unsigned char *user
 
 	return aes_set_decrypt_key_generic(userKey, bits, key);
 }
+#endif
 
+#ifdef HAVE_AES_ENCRYPT_INTERNAL
 void
 aes_encrypt_internal(const unsigned char *in, unsigned char *out,
     const AES_KEY *key)
@@ -76,7 +81,9 @@ aes_encrypt_internal(const unsigned char *in, unsigned
 
 	aes_encrypt_generic(in, out, key);
 }
+#endif
 
+#ifdef HAVE_AES_DECRYPT_INTERNAL
 void
 aes_decrypt_internal(const unsigned char *in, unsigned char *out,
     const AES_KEY *key)
@@ -88,7 +95,9 @@ aes_decrypt_internal(const unsigned char *in, unsigned
 
 	aes_decrypt_generic(in, out, key);
 }
+#endif
 
+#ifdef HAVE_AES_CBC_ENCRYPT_INTERNAL
 void
 aes_cbc_encrypt_internal(const unsigned char *in, unsigned char *out,
     size_t len, const AES_KEY *key, unsigned char *ivec, const int enc)
@@ -100,3 +109,4 @@ aes_cbc_encrypt_internal(const unsigned char *in, unsi
 
 	aes_cbc_encrypt_generic(in, out, len, key, ivec, enc);
 }
+#endif
