--- crypto/arch/amd64/crypto_arch.h.orig	Fri Oct 24 07:06:10 2025
+++ crypto/arch/amd64/crypto_arch.h	Fri Oct 24 07:06:21 2025
@@ -53,6 +53,7 @@ extern uint64_t crypto_cpu_caps_amd64;
 #define HAVE_RC4_INTERNAL
 #define HAVE_RC4_SET_KEY_INTERNAL
 
+#if 0
 #define HAVE_SHA1_BLOCK_DATA_ORDER
 #define HAVE_SHA1_BLOCK_GENERIC
 
@@ -61,6 +62,7 @@ extern uint64_t crypto_cpu_caps_amd64;
 
 #define HAVE_SHA512_BLOCK_DATA_ORDER
 #define HAVE_SHA512_BLOCK_GENERIC
+#endif
 
 #endif
 
