--- crypto/arch/amd64/crypto_arch.h.orig	Sun Aug 17 13:14:19 2025
+++ crypto/arch/amd64/crypto_arch.h	Sun Aug 24 23:47:56 2025
@@ -48,6 +48,7 @@ extern uint64_t crypto_cpu_caps_amd64;
 #define HAVE_RC4_INTERNAL
 #define HAVE_RC4_SET_KEY_INTERNAL
 
+#if 0
 #define HAVE_SHA1_BLOCK_DATA_ORDER
 #define HAVE_SHA1_BLOCK_GENERIC
 
@@ -56,6 +57,7 @@ extern uint64_t crypto_cpu_caps_amd64;
 
 #define HAVE_SHA512_BLOCK_DATA_ORDER
 #define HAVE_SHA512_BLOCK_GENERIC
+#endif
 
 #endif
 
