--- crypto/compat/arc4random.c.orig	2015-07-20 07:41:17.000000000 -0600
+++ crypto/compat/arc4random.c	2015-07-20 07:41:58.000000000 -0600
@@ -36,8 +36,11 @@
 #define KEYSTREAM_ONLY
 #include "chacha_private.h"
 
+#ifndef min
 #define min(a, b) ((a) < (b) ? (a) : (b))
-#ifdef __GNUC__
+#endif
+
+#if defined(__GNUC__) || defined(_MSC_VER)
 #define inline __inline
 #else				/* !__GNUC__ */
 #define inline
