--- tests/bn_isqrt.c.orig	Fri Dec  9 11:05:26 2022
+++ tests/bn_isqrt.c	Fri Dec  9 11:12:37 2022
@@ -306,6 +306,7 @@ main(int argc, char *argv[])
 	int ch;
 	int failed = 0, print = 0;
 
+#ifndef _MSC_VER
 	while ((ch = getopt(argc, argv, "C")) != -1) {
 		switch (ch) {
 		case 'C':
@@ -316,6 +317,7 @@ main(int argc, char *argv[])
 			break;
 		}
 	}
+#endif
 
 	if (print)
 		return check_tables(1);
