--- tests/bn_shift.c.orig	2023-02-13 20:06:27.295678033 -0600
+++ tests/bn_shift.c	2023-02-13 20:08:08.335677654 -0600
@@ -355,6 +355,8 @@
 	return failed;
 }
 
+#if 0
+
 static void
 benchmark_bn_lshift1(BIGNUM *bn)
 {
@@ -620,9 +622,12 @@
 	BN_free(bn);
 }
 
+#endif
+
 static void
 benchmark_bn_shift(void)
 {
+#if 0
 	const struct benchmark *bm;
 	size_t i;
 
@@ -630,6 +635,9 @@
 		bm = &benchmarks[i];
 		benchmark_run(bm, 5);
 	}
+#else
+	return;
+#endif
 }
 
 int
