--- tests/bn_shift.c.orig	Sun Dec 18 01:05:29 2022
+++ tests/bn_shift.c	Sun Dec 18 01:29:36 2022
@@ -314,6 +314,8 @@ test_bn_rshift_to_zero(void)
 	return failed;
 }
 
+#if 0
+
 static void
 benchmark_bn_lshift1(BIGNUM *bn)
 {
@@ -507,9 +509,12 @@ benchmark_run(const struct benchmark *bm, int seconds)
 	BN_free(bn);
 }
 
+#endif
+
 static void
 benchmark_bn_shift(void)
 {
+#if 0
 	const struct benchmark *bm;
 	size_t i;
 
@@ -517,6 +522,9 @@ benchmark_bn_shift(void)
 		bm = &benchmarks[i];
 		benchmark_run(bm, 5);
 	}
+#else
+	return;
+#endif
 }
 
 int
