--- crypto/arch/amd64/crypto_arch.h.orig	Fri Feb 14 06:00:43 2025
+++ crypto/arch/amd64/crypto_arch.h	Fri Feb 14 06:01:13 2025
@@ -40,6 +40,7 @@ extern uint64_t crypto_cpu_caps_amd64;
 #define HAVE_RC4_INTERNAL
 #define HAVE_RC4_SET_KEY_INTERNAL
 
+#if 0
 #define HAVE_SHA1_BLOCK_DATA_ORDER
 #define HAVE_SHA1_BLOCK_GENERIC
 
@@ -48,6 +49,7 @@ extern uint64_t crypto_cpu_caps_amd64;
 
 #define HAVE_SHA512_BLOCK_DATA_ORDER
 #define HAVE_SHA512_BLOCK_GENERIC
+#endif
 
 #endif
 
