--- crypto/hidden/crypto_namespace.h.orig	2023-02-28 00:41:43.212756628 -0600
+++ crypto/hidden/crypto_namespace.h	2023-02-28 00:42:01.152756349 -0600
@@ -18,6 +18,10 @@
 #ifndef _LIBCRYPTO_CRYPTO_NAMESPACE_H_
 #define _LIBCRYPTO_CRYPTO_NAMESPACE_H_
 
+#if defined(_MSC_VER) && !defined(__clang__) && !defined(__attribute__)
+#define __attribute__(a)
+#endif
+
 /*
  * If marked as 'used', then internal calls use the name with prefix "_lcry_"
  * and we alias that to the normal name *and* the name with prefix "_libre_";
