--- crypto/hidden/crypto_namespace.h.orig	Fri Aug  2 23:52:55 2024
+++ crypto/hidden/crypto_namespace.h	Fri Aug  2 23:53:17 2024
@@ -24,6 +24,12 @@
  * external calls use the latter name.
  */
 
+#ifdef _MSC_VER
+# define LCRYPTO_UNUSED(x)
+# define LCRYPTO_USED(x)
+# define LCRYPTO_ALIAS1(pre, x)
+# define LCRYPTO_ALIAS(x)
+#else
 #ifdef LIBRESSL_NAMESPACE
 #ifdef LIBRESSL_CRYPTO_NAMESPACE
 #  define LCRYPTO_UNUSED(x)	__attribute__((deprecated))		\
@@ -47,5 +53,6 @@
 # define LCRYPTO_ALIAS1(pre,x)
 # define LCRYPTO_ALIAS(x)	asm("")
 #endif
+#endif /* _MSC_VER */
 
 #endif	/* _LIBCRYPTO_CRYPTO_NAMESPACE_H_ */
