--- openbsd/src/lib/libcrypto/hidden/crypto_namespace.h	2023-03-15 11:41:37.614024500 -0600
+++ crypto/hidden/crypto_namespace.h	2023-03-15 17:12:55.297157400 -0600
@@ -24,6 +24,12 @@
  * external calls use the latter name.
  */
 
+#ifdef _MSC_VER
+# define LCRYPTO_UNUSED(x)
+# define LCRYPTO_USED(x)
+# define LCRYPTO_ALIAS1(pre, x)
+# define LCRYPTO_ALIAS(x)
+#else
 #ifdef LIBRESSL_NAMESPACE
 # define LCRYPTO_UNUSED(x)		typeof(x) x __attribute__((deprecated))
 #ifdef LIBRESSL_CRYPTO_NAMESPACE
@@ -40,5 +46,6 @@
 # define LCRYPTO_ALIAS1(pre,x)
 # define LCRYPTO_ALIAS(x)	asm("")
 #endif
+#endif /* _MSC_VER */
 
 #endif	/* _LIBCRYPTO_CRYPTO_NAMESPACE_H_ */
