--- crypto/modes/modes_local.h.orig	Sat Jul  8 14:03:53 2023
+++ crypto/modes/modes_local.h	Sat Jul  8 14:10:56 2023
@@ -45,6 +45,7 @@
 			asm ("bswapl %0"				\
 			: "+r"(ret));	ret;		})
 # elif (defined(__arm__) || defined(__arm)) && !defined(__STRICT_ALIGNMENT)
+#  if (__ARM_ARCH >= 6)
 #  define BSWAP8(x) ({	u32 lo=(u64)(x)>>32,hi=(x);			\
 			asm ("rev %0,%0; rev %1,%1"			\
 			: "+r"(hi),"+r"(lo));				\
@@ -53,6 +54,7 @@
 			asm ("rev %0,%1"				\
 			: "=r"(ret) : "r"((u32)(x)));			\
 			ret;				})
+#  endif
 # endif
 #endif
 #endif
