--- ssl/hidden/ssl_namespace.h.orig	Sun Aug 17 13:14:19 2025
+++ ssl/hidden/ssl_namespace.h	Mon Aug 18 04:11:53 2025
@@ -23,6 +23,10 @@
  * and we alias that to the normal name.
  */
 
+#ifdef _MSC_VER
+#undef LIBRESSL_NAMESPACE
+#endif
+
 #ifdef LIBRESSL_NAMESPACE
 #define LSSL_UNUSED(x)		typeof(x) x __attribute__((deprecated))
 #define LSSL_USED(x)		__attribute__((visibility("hidden")))	\
@@ -35,7 +39,11 @@
 #else
 #define LSSL_UNUSED(x)
 #define LSSL_USED(x)
+#ifdef _MSC_VER
+#define LSSL_ALIAS(x)
+#else
 #define LSSL_ALIAS(x)		asm("")
+#endif /* _MSC_VER */
 #endif
 
 #endif	/* _LIBSSL_SSL_NAMESPACE_H_ */
