--- ssl/hidden/ssl_namespace.h.orig	2023-02-28 00:42:20.928756042 -0600
+++ ssl/hidden/ssl_namespace.h	2023-02-28 00:43:16.372755180 -0600
@@ -18,6 +18,10 @@
 #ifndef _LIBSSL_SSL_NAMESPACE_H_
 #define _LIBSSL_SSL_NAMESPACE_H_
 
+#if defined(_MSC_VER) && !defined(__clang__) && !defined(__attribute__)
+#define __attribute__(a)
+#endif
+
 /*
  * If marked as 'used', then internal calls use the name with prefix "_lssl_"
  * and we alias that to the normal name.
