--- openbsd/src/lib/libssl/hidden/ssl_namespace.h	2023-03-15 11:41:52.481641800 -0600
+++ ssl/hidden/ssl_namespace.h	2023-03-15 17:28:04.685309300 -0600
@@ -23,6 +23,11 @@
  * and we alias that to the normal name.
  */
 
+#ifdef _MSC_VER
+#define LSSL_UNUSED(x)
+#define LSSL_USED(x)
+#define LSSL_ALIAS(x)
+#else
 #ifdef LIBRESSL_NAMESPACE
 #define LSSL_UNUSED(x)		typeof(x) x __attribute__((deprecated))
 #define LSSL_USED(x)		__attribute__((visibility("hidden")))	\
@@ -33,5 +38,6 @@
 #define LSSL_USED(x)
 #define LSSL_ALIAS(x)		asm("")
 #endif
+#endif /* _MSC_VER */
 
 #endif	/* _LIBSSL_SSL_NAMESPACE_H_ */
