--- ssl/hidden/ssl_namespace.h.orig	Fri Aug  2 23:52:55 2024
+++ ssl/hidden/ssl_namespace.h	Fri Aug  2 23:53:17 2024
@@ -23,6 +23,11 @@
  * and we alias that to the normal name.
  */
 
+#ifdef _MSC_VER
+#define LSSL_UNUSED(x)
+#define LSSL_USED(x)
+#define LSSL_ALIAS(x)
+#else
 #ifdef LIBRESSL_NAMESPACE
 #define LSSL_UNUSED(x)		typeof(x) x __attribute__((deprecated))
 #define LSSL_USED(x)		__attribute__((visibility("hidden")))	\
@@ -37,5 +42,6 @@
 #define LSSL_USED(x)
 #define LSSL_ALIAS(x)		asm("")
 #endif
+#endif /* _MSC_VER */
 
 #endif	/* _LIBSSL_SSL_NAMESPACE_H_ */
