--- include/tls.h.orig	2015-05-23 19:18:30.002576267 -0500
+++ include/tls.h	2015-05-23 19:18:09.830576581 -0500
@@ -18,6 +18,13 @@
 #ifndef HEADER_TLS_H
 #define HEADER_TLS_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stddef.h>
+#include <stdint.h>
+
 #define TLS_API	20141031
 
 #define TLS_PROTOCOL_TLSv1_0	(1 << 1)
@@ -88,4 +95,8 @@
 
 uint8_t *tls_load_file(const char *_file, size_t *_len, char *_password);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif /* HEADER_TLS_H */
