--- include/openssl/x509.h.orig	2015-07-06 13:15:15.059306046 -0700
+++ include/openssl/x509.h	2015-07-06 13:16:10.506118278 -0700
@@ -112,6 +112,19 @@
 extern "C" {
 #endif

+#if defined(_WIN32)
+#ifndef LIBRESSL_INTERNAL
+#ifdef _MSC_VER
+#pragma message("Warning, overriding WinCrypt defines")
+#else
+#warning overriding WinCrypt defines
+#endif
+#endif
+#undef X509_NAME
+#undef X509_CERT_PAIR
+#undef X509_EXTENSIONS
+#endif
+
 #define X509_FILETYPE_PEM	1
 #define X509_FILETYPE_ASN1	2
 #define X509_FILETYPE_DEFAULT	3
