diff -uprN -X mkfile apps.orig/openssl/apps.c apps/openssl/apps.c
--- apps.orig/openssl/apps.c	Sun May 31 21:08:21 2020
+++ apps/openssl/apps.c	Sun May 31 19:27:30 2020
@@ -146,6 +146,7 @@
 #include <openssl/x509v3.h>
 
 #include <openssl/rsa.h>
+#include <openssl/asn1t.h>
 
 typedef struct {
 	const char *name;
diff -uprN -X mkfile apps.orig/openssl/x509.c apps/openssl/x509.c
--- apps.orig/openssl/x509.c	Sun May 31 21:08:22 2020
+++ apps/openssl/x509.c	Sun May 31 20:33:49 2020
@@ -65,6 +65,7 @@
 #include "apps.h"
 
 #include <openssl/asn1.h>
+#include <openssl/asn1t.h>
 #include <openssl/bio.h>
 #include <openssl/bn.h>
 #include <openssl/err.h>
@@ -776,7 +777,7 @@ x509_main(int argc, char **argv)
 			} else if (startdate == i) {
 				ASN1_TIME *nB = X509_get_notBefore(x);
 				BIO_puts(STDout, "notBefore=");
-				if (ASN1_time_parse(nB->data, nB->length, NULL, 0) == -1)
+				if (ASN1_time_parse((const char *)nB->data, nB->length, NULL, 0) == -1)
 					BIO_puts(STDout, "INVALID RFC5280 TIME");
 				else
 					ASN1_TIME_print(STDout, nB);
