diff -uprN -X mkfile crypto.orig/chacha/chacha-merged.c crypto/chacha/chacha-merged.c
--- crypto.orig/chacha/chacha-merged.c	Sun May 31 21:08:04 2020
+++ crypto/chacha/chacha-merged.c	Sun May 31 19:27:31 2020
@@ -7,6 +7,7 @@ Public domain.
 
 #include <sys/types.h>
 
+#include <stddef.h>
 #include <stdint.h>
 
 #define CHACHA_MINKEYLEN 	16
diff -uprN -X mkfile crypto.orig/cryptlib.c crypto/cryptlib.c
--- crypto.orig/cryptlib.c	Sun May 31 21:07:59 2020
+++ crypto/cryptlib.c	Sun May 31 19:27:31 2020
@@ -119,6 +119,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <syslog.h>
+#include <sys/types.h>
 #include <unistd.h>
 
 #include <openssl/opensslconf.h>
diff -uprN -X mkfile crypto.orig/ocsp/ocsp_cl.c crypto/ocsp/ocsp_cl.c
--- crypto.orig/ocsp/ocsp_cl.c	Sun May 31 21:08:12 2020
+++ crypto/ocsp/ocsp_cl.c	Sun May 31 19:27:31 2020
@@ -70,6 +70,7 @@
 #include <openssl/pem.h>
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 /* Utility functions related to sending OCSP requests and extracting
  * relevant information from the response.
diff -uprN -X mkfile crypto.orig/ocsp/ocsp_srv.c crypto/ocsp/ocsp_srv.c
--- crypto.orig/ocsp/ocsp_srv.c	Sun May 31 21:08:12 2020
+++ crypto/ocsp/ocsp_srv.c	Sun May 31 19:27:31 2020
@@ -64,6 +64,7 @@
 #include <openssl/pem.h>
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 /* Utility functions related to sending OCSP responses and extracting
  * relevant information from the request.
diff -uprN -X mkfile crypto.orig/ocsp/ocsp_vfy.c crypto/ocsp/ocsp_vfy.c
--- crypto.orig/ocsp/ocsp_vfy.c	Sun May 31 21:08:12 2020
+++ crypto/ocsp/ocsp_vfy.c	Sun May 31 19:27:32 2020
@@ -58,6 +58,7 @@
 
 #include <openssl/ocsp.h>
 #include <openssl/err.h>
+#include <openssl/asn1t.h>
 #include <string.h>
 
 static int ocsp_find_signer(X509 **psigner, OCSP_BASICRESP *bs,
diff -uprN -X mkfile crypto.orig/pkcs12/p12_add.c crypto/pkcs12/p12_add.c
--- crypto.orig/pkcs12/p12_add.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs12/p12_add.c	Sun May 31 19:27:32 2020
@@ -60,6 +60,7 @@
 
 #include <openssl/err.h>
 #include <openssl/pkcs12.h>
+#include <openssl/asn1t.h>
 
 /* Pack an object into an OCTET STRING and turn into a safebag */
 
diff -uprN -X mkfile crypto.orig/pkcs12/p12_p8d.c crypto/pkcs12/p12_p8d.c
--- crypto.orig/pkcs12/p12_p8d.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs12/p12_p8d.c	Sun May 31 19:27:32 2020
@@ -59,6 +59,7 @@
 #include <stdio.h>
 
 #include <openssl/pkcs12.h>
+#include <openssl/asn1t.h>
 
 PKCS8_PRIV_KEY_INFO *
 PKCS8_decrypt(const X509_SIG *p8, const char *pass, int passlen)
diff -uprN -X mkfile crypto.orig/pkcs12/p12_p8e.c crypto/pkcs12/p12_p8e.c
--- crypto.orig/pkcs12/p12_p8e.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs12/p12_p8e.c	Sun May 31 19:27:32 2020
@@ -60,6 +60,7 @@
 
 #include <openssl/err.h>
 #include <openssl/pkcs12.h>
+#include <openssl/asn1t.h>
 
 X509_SIG *
 PKCS8_encrypt(int pbe_nid, const EVP_CIPHER *cipher, const char *pass,
diff -uprN -X mkfile crypto.orig/pkcs12/p12_utl.c crypto/pkcs12/p12_utl.c
--- crypto.orig/pkcs12/p12_utl.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs12/p12_utl.c	Sun May 31 19:27:32 2020
@@ -61,6 +61,7 @@
 #include <string.h>
 
 #include <openssl/pkcs12.h>
+#include <openssl/asn1t.h>
 
 /* Cheap and nasty Unicode stuff */
 
diff -uprN -X mkfile crypto.orig/pkcs7/bio_pk7.c crypto/pkcs7/bio_pk7.c
--- crypto.orig/pkcs7/bio_pk7.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs7/bio_pk7.c	Sun May 31 19:27:32 2020
@@ -53,6 +53,7 @@
  */
 
 #include <openssl/asn1.h>
+#include <openssl/asn1t.h>
 #include <openssl/pkcs7.h>
 #include <openssl/bio.h>
 
diff -uprN -X mkfile crypto.orig/pkcs7/pk7_doit.c crypto/pkcs7/pk7_doit.c
--- crypto.orig/pkcs7/pk7_doit.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs7/pk7_doit.c	Sun May 31 19:27:32 2020
@@ -64,6 +64,7 @@
 #include <openssl/objects.h>
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 static int add_attribute(STACK_OF(X509_ATTRIBUTE) **sk, int nid, int atrtype,
     void *value);
diff -uprN -X mkfile crypto.orig/pkcs7/pk7_mime.c crypto/pkcs7/pk7_mime.c
--- crypto.orig/pkcs7/pk7_mime.c	Sun May 31 21:08:13 2020
+++ crypto/pkcs7/pk7_mime.c	Sun May 31 19:27:32 2020
@@ -57,6 +57,7 @@
 
 #include <openssl/asn1.h>
 #include <openssl/x509.h>
+#include <openssl/asn1t.h>
 
 /* PKCS#7 wrappers round generalised stream and MIME routines */
 
diff -uprN -X mkfile crypto.orig/poly1305/poly1305-donna.c crypto/poly1305/poly1305-donna.c
--- crypto.orig/poly1305/poly1305-donna.c	Sun May 31 21:08:13 2020
+++ crypto/poly1305/poly1305-donna.c	Sun May 31 19:27:32 2020
@@ -6,6 +6,7 @@
  */
 
 #include <stddef.h>
+#include <openssl/poly1305.h>
 
 static inline void poly1305_init(poly1305_context *ctx,
     const unsigned char key[32]);
diff -uprN -X mkfile crypto.orig/x509/x_all.c crypto/x509/x_all.c
--- crypto.orig/x509/x_all.c	Sun May 31 21:08:16 2020
+++ crypto/x509/x_all.c	Sun May 31 19:27:32 2020
@@ -61,6 +61,7 @@
 #include <openssl/opensslconf.h>
 
 #include <openssl/asn1.h>
+#include <openssl/asn1t.h>
 #include <openssl/buffer.h>
 #include <openssl/evp.h>
 #include <openssl/stack.h>
diff -uprN -X mkfile crypto.orig/x509v3/v3_alt.c crypto/x509v3/v3_alt.c
--- crypto.orig/x509v3/v3_alt.c	Sun May 31 21:08:16 2020
+++ crypto/x509v3/v3_alt.c	Sun May 31 19:27:32 2020
@@ -62,6 +62,7 @@
 #include <openssl/conf.h>
 #include <openssl/err.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 static GENERAL_NAMES *v2i_subject_alt(X509V3_EXT_METHOD *method,
     X509V3_CTX *ctx, STACK_OF(CONF_VALUE) *nval);
diff -uprN -X mkfile crypto.orig/x509v3/v3_bitst.c crypto/x509v3/v3_bitst.c
--- crypto.orig/x509v3/v3_bitst.c	Sun May 31 21:08:16 2020
+++ crypto/x509v3/v3_bitst.c	Sun May 31 19:27:33 2020
@@ -62,6 +62,7 @@
 #include <openssl/conf.h>
 #include <openssl/err.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 static BIT_STRING_BITNAME ns_cert_type_table[] = {
 	{0, "SSL Client", "client"},
diff -uprN -X mkfile crypto.orig/x509v3/v3_enum.c crypto/x509v3/v3_enum.c
--- crypto.orig/x509v3/v3_enum.c	Sun May 31 21:08:16 2020
+++ crypto/x509v3/v3_enum.c	Sun May 31 19:27:33 2020
@@ -59,6 +59,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 static ENUMERATED_NAMES crl_reasons[] = {
 	{CRL_REASON_UNSPECIFIED, 	 "Unspecified", "unspecified"},
diff -uprN -X mkfile crypto.orig/x509v3/v3_ia5.c crypto/x509v3/v3_ia5.c
--- crypto.orig/x509v3/v3_ia5.c	Sun May 31 21:08:17 2020
+++ crypto/x509v3/v3_ia5.c	Sun May 31 19:27:33 2020
@@ -60,6 +60,7 @@
 #include <string.h>
 
 #include <openssl/asn1.h>
+#include <openssl/asn1t.h>
 #include <openssl/conf.h>
 #include <openssl/err.h>
 #include <openssl/x509v3.h>
diff -uprN -X mkfile crypto.orig/x509v3/v3_int.c crypto/x509v3/v3_int.c
--- crypto.orig/x509v3/v3_int.c	Sun May 31 21:08:17 2020
+++ crypto/x509v3/v3_int.c	Sun May 31 19:27:33 2020
@@ -59,6 +59,7 @@
 #include <stdio.h>
 
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 const X509V3_EXT_METHOD v3_crl_num = {
 	.ext_nid = NID_crl_number,
diff -uprN -X mkfile crypto.orig/x509v3/v3_ocsp.c crypto/x509v3/v3_ocsp.c
--- crypto.orig/x509v3/v3_ocsp.c	Sun May 31 21:08:17 2020
+++ crypto/x509v3/v3_ocsp.c	Sun May 31 19:27:33 2020
@@ -64,6 +64,7 @@
 #ifndef OPENSSL_NO_OCSP
 
 #include <openssl/asn1.h>
+#include <openssl/asn1t.h>
 #include <openssl/conf.h>
 #include <openssl/err.h>
 #include <openssl/ocsp.h>
diff -uprN -X mkfile crypto.orig/x509v3/v3_pci.c crypto/x509v3/v3_pci.c
--- crypto.orig/x509v3/v3_pci.c	Sun May 31 21:08:17 2020
+++ crypto/x509v3/v3_pci.c	Sun May 31 19:27:33 2020
@@ -40,6 +40,7 @@
 #include <openssl/conf.h>
 #include <openssl/err.h>
 #include <openssl/x509v3.h>
+#include <openssl/asn1t.h>
 
 static int i2r_pci(X509V3_EXT_METHOD *method, PROXY_CERT_INFO_EXTENSION *ext,
     BIO *out, int indent);
diff -uprN -X mkfile crypto.orig/x509v3/v3_skey.c crypto/x509v3/v3_skey.c
--- crypto.orig/x509v3/v3_skey.c	Sun May 31 21:08:17 2020
+++ crypto/x509v3/v3_skey.c	Sun May 31 19:27:33 2020
@@ -60,6 +60,7 @@
 #include <string.h>
 
 #include <openssl/err.h>
+#include <openssl/asn1t.h>
 #include <openssl/x509v3.h>
 
 static ASN1_OCTET_STRING *s2i_skey_id(X509V3_EXT_METHOD *method,
diff -uprN -X mkfile crypto.orig/x509v3/v3_utl.c crypto/x509v3/v3_utl.c
--- crypto.orig/x509v3/v3_utl.c	Sun May 31 21:08:17 2020
+++ crypto/x509v3/v3_utl.c	Sun May 31 19:27:33 2020
@@ -61,6 +61,7 @@
 #include <stdio.h>
 #include <string.h>
 
+#include <openssl/asn1t.h>
 #include <openssl/bn.h>
 #include <openssl/conf.h>
 #include <openssl/err.h>
